Spice/src/Config/UserAgent.cs

```diff
@@ -24,7 +24,7 @@ namespace Spice.Config;
 
 internal static class UserAgent
 {
-    public static string agent()
+    public static string agent(string? client = null, string? clientVersion = null, string? clientSystem = null, string? clientExtension = null)
```

Comment:
Nit: might be good to make it public so users can benefit and rename it to `SpiceUserAgent`